src/runtime/link-accounts-flow.js

```diff
@@ -223,25 +233,44 @@ export class LinkSaveFlow {
       'isClosable': true,
     };
 
-    if (this.request_.token) {
-      if (!this.request_.authCode) {
-        iframeArgs['token'] = this.request_.token;
-      } else {
-        throw new Error('Both authCode and token are available');
-      }
-    } else if (this.request_.authCode) {
-      iframeArgs['authCode'] = this.request_.authCode;
-    } else {
-      throw new Error('Neither token or authCode is available');
-    }
-
     this.activityIframeView_ = new ActivityIframeView(
       this.win_,
       this.activityPorts_,
       feUrl('/linksaveiframe'),
       feArgs(iframeArgs),
       /* shouldFadeBody */ false
     );
+
+    this.activityIframeView_.onMessage(data => {
+      if (data['getLinkingInfo']) {
+        /** {!Promise<../api/subscriptions.SaveSubscriptionRequest>} */
+        this.requestPromise_ = new Promise(resolve => {
+          resolve(this.callback_());
+        }).then(request => {
+          if (!request) {
+            throw new Error('Neither token or authCode is available');
```

Comment:
Duplicated from below. You can do it this way:

```
if (request && request.token) {
} else if (request && request.authCode)  {
} else {
  throw new Error('Both ...');
}
```

Reply:
done